bin/critterget.py

```diff
@@ -563,19 +563,23 @@ def getCredentials(sessionKey):
 
     if (debug) : print u'{} MessageType="ApteligentDebug"  Into getCredentials'.format(DATETIME_OF_RUN)
 
-    try:
-        # list all credentials
-        entities = entity.getEntities(['admin', 'passwords'], namespace=myapp,
-                                    owner='nobody', sessionKey=sessionKey)
-    except Exception, e:
-        print u'{} MessageType="ApteligentDebug" Could not get {} credentials from splunk. Error: {}'.format(DATETIME_OF_RUN, myapp, str(e))
-
-    # return first set of credentials
-    if (debug) : print "Entities is ", entities
-    for i, c in entities.items():
-        return c['clear_password']
-
-    print u'{} MessageType="ApteligentDebug" No credentials have been found for app {} . Maybe a setup issue?'.format(DATETIME_OF_RUN, myapp)
+    auth = None
+
+    while auth is None:
+        try:
+            # list all credentials
+            entities = entity.getEntities(['admin', 'passwords'], namespace=myapp,
```

Comment:
Ok, so this try/catch here is sort of useless: if an exception is thrown, `entities` will be undefined and out of scope.  The code block below will throw a syntax error when it tries to print it.

Assign `entities` to a reasonable default, like `None` or an empty dictionary, before the try block.